Using createUndoRedoReducer with StoreModule.forFeature doesn't work

I followed the installation guide created a reducer for a feature instead of root and it didn't work, not getting any errors as well other than the one below, which is just error because the state doesn't get initialized and the selectors are trying to access the data.(ngrx-wieder tries to access the store as well and fails).

Doing nothing fancy here, just trying to create a reducer and use it with `StoreModule.forFeature('databoard', undoableRedcucer)`

If I use createReducer from `ngrx/store` It works fine. 

```sh
error-handler.service.ts:23 Error: TypeError: Cannot read property 'histories' of undefined
    at ngrx-wieder.js:88
    at reducer (databoard.reducer.ts:289)
    at ngrx-store.js:290
    at combination (ngrx-store.js:245)
    at ngrx-store.js:752
    at ngrx-store.js:752
    at ngrx-store.js:278
    at computeNextEntry (ngrx-store-devtools.js:426)
    at recomputeStates (ngrx-store-devtools.js:460)
    at ngrx-store-devtools.js:734

ngrx-store.js:695 @ngrx/store: The feature name "databoard" does not exist in the state, therefore createFeatureSelector cannot access it.  Be sure it is imported in a loaded module using StoreModule.forRoot('databoard', ...) or StoreModule.forFeature('databoard', ...).  If the default state is intended to be undefined, as is the case with router state, this development-only warning message can be ignored.
(anonymous) @ ngrx-store.js:695
(anonymous) @ ngrx-store.js:571
defaultStateFn @ ngrx-store.js:571

databoard.reducer.ts:319 selectorState undefined
error-handler.service.ts:23 Error: TypeError: Cannot read property 'histories' of undefined
    at ngrx-wieder.js:88
    at reducer (databoard.reducer.ts:289)
    at ngrx-store.js:290
    at combination (ngrx-store.js:245)
    at ngrx-store.js:752
    at ngrx-store.js:752
    at ngrx-store.js:278
    at ngrx-store-devtools.js:737
    at Array.map (<anonymous>)
    at ngrx-store-devtools.js:737
```

--- a/projects/ngrx-wieder/src/lib/integration.spec.ts
+++ b/projects/ngrx-wieder/src/lib/integration.spec.ts
@@ -1,0 +1,51 @@
+import { TestBed } from "@angular/core/testing";
+import { ActionReducer, Action, StoreModule, Store } from "@ngrx/store";
+import {
+  TestState,
+  initialState,
+  createOnReducer,
+  createSwitchReducer,
+} from "./test-util/store";
+
+interface RootState {
+  rootSlice: TestState;
+  feat: {
+    featSlice: TestState;
+  };
+}
+
+const test = (createReducer: () => ActionReducer<TestState, Action>) => {
+  let store: Store<RootState>;
+  beforeEach(() => {
+    const rootReducers = {
+      rootSlice: createReducer(),
+    };
+    const featReducers = {
+      featSlice: createReducer(),
+    };
+    TestBed.configureTestingModule({
+      imports: [
+        StoreModule.forRoot(rootReducers),
+        StoreModule.forFeature("feat", featReducers),
+      ],
+    });
+    store = TestBed.inject(Store);
+  });
+
+  it("should startup store", (done) => {
+    store.subscribe((state) => {
+      expect(state).toEqual({
+        rootSlice: initialState,
+        feat: {
+          featSlice: initialState,
+        },
+      });
+      done();
+    });
+  });
+};
+
+describe("Integration", () => {
+  describe("OnReducer", () => test(createOnReducer));
+  describe("SwitchReducer", () => test(createSwitchReducer));
+});

--- a/projects/ngrx-wieder/src/lib/undo-redo.reducer.ts
+++ b/projects/ngrx-wieder/src/lib/undo-redo.reducer.ts
@@ -4,6 +4,7 @@ import {
   ActionReducer,
   INIT,
   ReducerTypes,
+  UPDATE,
 } from "@ngrx/store";
 import produce, { applyPatches, enablePatches, PatchListener } from "immer";
 import {
@@ -132,7 +133,7 @@ function wrap<S extends UndoRedoState, A extends Action = Action>(
   };
 
   return (state: S, action: A): S => {
-    if (state == null) {
+    if (action.type === INIT || action.type === UPDATE) {
       return reducer(state, action);
     }
     const key = segmentationKey(state, action);

--- a/projects/ngrx-wieder/src/lib/undo-redo.reducer.ts
+++ b/projects/ngrx-wieder/src/lib/undo-redo.reducer.ts
@@ -132,8 +132,7 @@ function wrap<S extends UndoRedoState, A extends Action = Action>(
   };
 
   return (state: S, action: A): S => {
-    if (action.type === INIT) {
-      // let reducer initialize state
+    if (state == null) {
       return reducer(state, action);
     }
     const key = segmentationKey(state, action);

